.github/workflows/cypress-tests.yml

```diff
@@ -9,19 +9,25 @@ on:
 jobs:
   cypress-run:
     runs-on: ubuntu-latest
-
+    strategy:
+      fail-fast: false
+      matrix:
+        containers: [0, 1, 2, 3, 4, 5, 6, 7]
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
-     
+
       - name: Use Node.js 16.x
         uses: actions/setup-node@v3
         with:
           node-version: 16.x
 
-
       - name: Install dependencies
         run: npm install
 
       - name: Cypress Tests are running
-        run : node  ./scripts/index.js && npm run test
+        uses: nick-fields/retry@v2
```

Comment:
how often timeouts happen?
I'm a bit afraid that default rerun for tests, will take over GH infra too much.

also I'm pretty sure cypress have native support for retry and timeout

Reply:
@AtharvChandratre Yeah, you can try cypress retry and timeout by adding it to the cypress config. Reference - https://docs.cypress.io/guides/guides/test-retries#Configure-Test-Retries

Reply:
@derberg thank you for the review. The suggestion makes complete sense. The timeout is in place because, as you pointed out, Cypress might keep running taking over a GH runner for too long. I looked into the default wait times for various cypress commands (https://docs.cypress.io/guides/core-concepts/introduction-to-cypress#Default-Values) and they seem to be sufficiently large for our use case, so I'm getting rid of the time out part.